### PR TITLE
platform(general): Add no upload flag and report contributors for all API key runs

### PR DESCRIPTION
**Pull request by mikeurbanski1**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Adds a flag to skip uploading results to the platform, regardless of source type. Source type upload defaults (e.g. true for github actions, false for IDE) still apply, but this can override any that is true.

The reason we avoided that flag until now is that we didn't want to make it trivial to use the platform "for free" without uploading any results. Now with dev pricing, that is not a concern, and so this PR also changes logic to report contributors to the platform for all runs with an API key, regardless of source and whether the results are uploaded. Thus, the billing and whether or not results get uploaded are decoupled for anyone on dev pricing (and we can live with it for anyone still on resource pricing who happens to use this).

Draft for now. There is a platform change required to not actually save contributors if the repo is public.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/util/ext_argument_parser.py [hunk 1]
```diff
@@ -310,6 +310,12 @@ def add_parser_args(self) -> None:
             help="The Prisma Cloud API URL (see: https://prisma.pan.dev/api/cloud/api-urls). "
                  "Requires --bc-api-key to be a Prisma Cloud Access Key in the following format: <access_key_id>::<secret_key>",
         )
+        self.add(
+            "--skip-results-upload",
+            action='store_true',
+            help="Do not upload scan results to the platform to view in the console. Results are only available locally. "
+                 "If you use the --support flag, logs will still get uploaded.",
```
Comment (tsmithv11) [inline checkov/common/util/ext_argument_parser.py:317]: 👍

File: docs/2.Basics/CLI Command Reference.md [hunk 2]
```diff
@@ -34,6 +34,7 @@ nav_order: 2
...
...                                                                                                                                                                                                                                                                | Do not upload scan results to the platform. Use this to download configs, but only view results in the local output.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
```
Comment (tsmithv11) [inline docs/2.Basics/CLI Command Reference.md:37]: 👍

**Review by gruebel — APPROVED** [review]
nice work 💪 should we also add the flag to the GHA directly?

**Comment by mikeurbanski1** [comment]
> nice work 💪 should we also add the flag to the GHA directly?

I think we should wait a month until someone asks about it, and then go "oh shit!"

(thanks for the reminder - I will draft those PRs)

**Comment by mikeurbanski1** [comment]
I created these:

https://github.com/bridgecrewio/checkov-action/pull/137

https://github.com/bridgecrewio/bridgecrew-action/pull/45

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**